File: cliente/src/js/service/person.service.js
(function () {
    'use strict';
    angular.module('angularStartApp').factory('PersonService', service);

    function service($http) {
        const PORT = 'http://localhost:8080/';
        const URL = PORT + 'test/';

        function list(filter) {
            return $http.get(URL + 'pessoa/list', {
                params: {
                    offset : filter.offset * filter.limit,
                    limit : filter.limit,
                }
            });
        }

        function save(pessoa) {
            return $http.post(URL + 'pessoa/save', pessoa);
        }

        function edit(pessoa) {
            return $http.post(URL + 'pessoa/edit', pessoa);
        }

        function remove(id) {
            return $http.delete(URL + 'pessoa/remove/' + id);
        }

        return {
            list: list,
            save: save,
            edit: edit,
            remove: remove
        };
    }
})();

File: servico/src/main/java/com/example/test/service/PersonService.java
package com.example.test.service;

import com.example.test.model.Person;
import com.example.test.model.dto.PersonDTO;
import com.example.test.repository.PersonRepository;
import com.example.test.util.MapperUtils;
import com.example.test.util.Resposta;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

import java.util.ArrayList;
import java.util.List;

@Service
public class PersonService {

    @Autowired
    private PersonRepository personRepository;

    @Transactional(propagation=Propagation.REQUIRED)
    public Resposta<List<PersonDTO>> list(int offset, int limit) {
        List<Person> personList = personRepository.findAllPaginado(offset, limit);
        Resposta<List<PersonDTO>> resposta = new Resposta<>();
        resposta.setResult(MapperUtils.map(personList, PersonDTO.class));
        resposta.setTotalElements(personRepository.countAll());

        return resposta;
    }

    @Transactional(propagation=Propagation.REQUIRED)
    public void save(PersonDTO personDTO) {
        Person person = new Person();
        person.setFirstName(personDTO.getFirstName());
        person.setLastName(personDTO.getLastName());
        personRepository.save(person);
    }

    @Transactional(propagation=Propagation.REQUIRED)
    public void edit(PersonDTO personDTO) {
        Person person = personRepository.findById(personDTO.getId()).get();

        person.setFirstName(personDTO.getFirstName());
        person.setLastName(personDTO.getLastName());
        personRepository.save(person);
    }

    @Transactional(propagation=Propagation.REQUIRED)
    public void remove(Long id) {
        if (id != null) {
            personRepository.delete(personRepository.findById(id).get());
        }
    }
}
